setup.py

```diff
@@ -25,48 +25,60 @@ def read(*filenames, **kwargs):
 
 root = os.path.dirname(os.path.realpath(__file__))
 version_module = imp.load_source(
-    'version', os.path.join(root, 'nengo_spa', 'version.py'))
-testing = 'test' in sys.argv or 'pytest' in sys.argv
+    "version", os.path.join(root, "nengo_spa", "version.py"))
+testing = "test" in sys.argv or "pytest" in sys.argv
+
+docs_require = [
+    "sphinx",
+    "nbsphinx",
+    "sphinx_rtd_theme",
+]
+optional_requires = [
+    "scipy"
+]
+tests_require = [
+    "jupyter",
+    "matplotlib>=2.0",
+    "pytest>=2.3",
+]
 
 setup(
     name="nengo_spa",
     version=version_module.version,
     author="Applied Brain Research",
     author_email="info@appliedbrainresearch.com",
-    packages=find_packages(exclude=['*.tests']),
+    packages=find_packages(exclude=["*.tests"]),
     scripts=[],
     license="Free for non-commercial use",
     description="An implementation of the Semantic Pointer Architecture for "
                 "Nengo",
-    long_description=read('README.rst', 'CHANGES.rst'),
+    long_description=read("README.rst", "CHANGES.rst"),
     zip_safe=True,
     include_package_data=True,
-    setup_requires=['pytest-runner'] if testing else [] + [
-        'numpy>=1.8',
+    setup_requires=["pytest-runner"] if testing else [] + [
+        "numpy>=1.8",
     ],
-    install_requires=[
-        'nengo>=2.7',
-        'numpy>=1.8',
+    install_requires = [
+        "nengo>=2.7",
+        "numpy>=1.8",
     ],
-    extras_require={
-        'prob_cleanup': ['scipy'],
+    extras_require = {
+        "all": docs_require + optional_requires + tests_require,
+        "docs": docs_require,
+        "optional": optional_requires,
+        "tests": tests_require
```

Comment:
Why are we no longer using the `tests_require` keyword argument?

Reply:
You should probably list test requirements in both an `extras_require` and `tests_require`, though note that the `tests_require` only comes into play if you do `python setup.py test` (and I think most of us don't do that).